Support naming customization on helm chart resources (#31066)
mistercrunch/Airflow: chart/newsfragments/31066.significant.rst

Support naming customization on helm chart resources, some resources may be renamed during upgrade

> This is a new opt-in switch ``useStandardNaming`` for backwards compatibility to leverage the standard naming convention and being able to use fully fullnameOverride and nameOverride in all resources.

Reviewer: ```suggestion
This is a new opt-in switch ``useStandardNaming``, for backwards compatibility, to leverage the standard naming convention, which allows full use of fullnameOverride and nameOverride in all resources.
```

> Only the following resources will be renamed using defaults ``useStandardNaming=false``:

Reviewer: ```suggestion
Only the following resources will be renamed using default of ``useStandardNaming=false``:
```
